### DISCUSSION TITLE
How to add custom icons

### DISCUSSION THREAD

**Discussion by victorbrca** [mayurankv/Obsidian-Code-Styler#185] [Q&A]
I'm trying to add a custom icon for bash but I'm not able to. I have exported the path for the following SVG from Gimp (and tried a few others), but none seem to work. 

https://upload.wikimedia.org/wikipedia/commons/thumb/4/4b/Bash_Logo_Colored.svg/2048px-Bash_Logo_Colored.svg.png

I can confirm that the path works - [see here](https://svg-path-visualizer.netlify.app/#M%2089.40%2C81.10%20C%2089.70%2C80.90%2089.90%2C81.10%2090.00%2C81.50%2090.00%2C81.50%2090.00%2C84.70%2090.00%2C84.70%2091.30%2C84.20%2092.50%2C84.00%2093.60%2C84.30%2093.80%2C84.40%2093.90%2C84.70%2093.80%2C85.00%2093.80%2C85.00%2093.10%2C87.80%2093.10%2C87.80%2093.00%2C88.00%2092.90%2C88.20%2092.80%2C88.40%2092.70%2C88.50%2092.70%2C88.50%2092.60%2C88.50%2092.50%2C88.50%2092.40%2C88.60%2092.30%2C88.50%2091.80%2C88.40%2090.70%2C88.10%2088.90%2C89.10%2087.00%2C90.10%2086.30%2C91.70%2086.40%2C92.90%2086.40%2C94.40%2087.20%2C94.80%2089.70%2C94.80%2093.10%2C94.90%2094.60%2C96.40%2094.70%2C99.80%2094.70%2C103.20%2092.90%2C106.90%2090.10%2C109.10%2090.10%2C109.10%2090.20%2C112.30%2090.20%2C112.30%2090.20%2C112.70%2090.00%2C113.10%2089.70%2C113.30%2089.70%2C113.30%2087.80%2C114.40%2087.80%2C114.40%2087.50%2C114.60%2087.30%2C114.40%2087.20%2C114.00%2087.20%2C114.00%2087.20%2C110.90%2087.20%2C110.90%2085.60%2C111.60%2084.00%2C111.70%2082.90%2C111.30%2082.70%2C111.20%2082.60%2C110.90%2082.70%2C110.60%2082.70%2C110.60%2083.40%2C107.70%2083.40%2C107.70%2083.50%2C107.50%2083.60%2C107.20%2083.70%2C107.10%2083.80%2C107.00%2083.80%2C107.00%2083.90%2C107.00%2084.00%2C106.90%2084.10%2C106.90%2084.20%2C107.00%2085.30%2C107.40%2086.80%2C107.20%2088.10%2C106.50%2089.90%2C105.60%2091.00%2C103.80%2091.00%2C102.00%2091.00%2C100.40%2090.10%2C99.70%2088.00%2C99.70%2085.30%2C99.70%2082.80%2C99.20%2082.70%2C95.20%2082.70%2C91.90%2084.40%2C88.50%2087.10%2C86.40%2087.10%2C86.40%2087.10%2C83.20%2087.10%2C83.20%2087.10%2C82.80%2087.30%2C82.40%2087.60%2C82.20%2087.60%2C82.20%2089.40%2C81.10%2089.40%2C81.10%20Z%20M%20109.20%2C101.50%20C%20109.20%2C101.50%2099.90%2C107.00%2099.90%2C107.00%2099.70%2C107.10%2099.50%2C107.30%2099.50%2C107.60%2099.50%2C107.60%2099.50%2C110.00%2099.50%2C110.00%2099.50%2C110.30%2099.70%2C110.40%2099.90%2C110.30%2099.90%2C110.30%20109.30%2C104.60%20109.30%2C104.60%20109.50%2C104.50%20109.60%2C104.20%20109.60%2C103.90%20109.60%2C103.90%20109.60%2C101.80%20109.60%2C101.80%20109.60%2C101.50%20109.40%2C101.40%20109.20%2C101.50%20Z%20M%20118.90%2C40.30%20C%20118.90%2C40.30%2081.70%2C18.20%2081.70%2C18.20%2079.50%2C16.90%2077.00%2C16.20%2074.50%2C16.20%2072.00%2C16.20%2069.50%2C16.90%2067.30%2C18.20%2067.30%2C18.20%2030.10%2C40.30%2030.10%2C40.30%2025.70%2C42.90%2022.90%2C47.80%2022.90%2C53.10%2022.90%2C53.10%2022.90%2C97.30%2022.90%2C97.30%2022.90%2C102.60%2025.60%2C107.40%2030.10%2C110.10%2030.10%2C110.10%2067.30%2C132.20%2067.30%2C132.20%2069.50%2C133.50%2072.00%2C134.20%2074.50%2C134.20%2077.00%2C134.20%2079.50%2C133.50%2081.70%2C132.20%2081.70%2C132.20%20118.90%2C110.10%20118.90%2C110.10%20123.30%2C107.50%20126.10%2C102.60%20126.10%2C97.30%20126.10%2C97.30%20126.10%2C53.00%20126.10%2C53.00%20126.10%2C47.80%20123.40%2C42.90%20118.90%2C40.30%20Z%20M%2068.60%2C129.80%20C%2068.60%2C129.80%2031.40%2C107.70%2031.40%2C107.70%2027.80%2C105.50%2025.50%2C101.50%2025.50%2C97.20%2025.50%2C97.20%2025.50%2C53.00%2025.50%2C53.00%2025.50%2C48.70%2027.80%2C44.60%2031.40%2C42.50%2031.40%2C42.50%2068.60%2C20.40%2068.60%2C20.40%2070.40%2C19.30%2072.40%2C18.80%2074.50%2C18.80%2076.60%2C18.80%2078.60%2C19.40%2080.40%2C20.40%2080.40%2C20.40%20117.60%2C42.50%20117.60%2C42.50%20120.70%2C44.30%20122.70%2C47.50%20123.30%2C51.00%20122.10%2C48.40%20119.30%2C47.60%20116.00%2C49.50%20116.00%2C49.50%2080.90%2C71.30%2080.90%2C71.30%2076.50%2C73.90%2073.30%2C76.70%2073.30%2C82.00%2073.30%2C82.00%2073.30%2C125.40%2073.30%2C125.40%2073.30%2C128.60%2074.60%2C130.60%2076.50%2C131.20%2075.90%2C131.30%2075.20%2C131.40%2074.50%2C131.40%2072.50%2C131.50%2070.40%2C130.90%2068.60%2C129.80%20Z%20M%20118.90%2C40.30%20C%20118.90%2C40.30%2081.70%2C18.20%2081.70%2C18.20%2077.30%2C15.60%2071.80%2C15.60%2067.30%2C18.20%2067.30%2C18.20%2030.10%2C40.30%2030.10%2C40.30%2025.70%2C42.90%2022.90%2C47.80%2022.90%2C53.10%2022.90%2C53.10%2022.90%2C97.30%2022.90%2C97.30%2022.90%2C102.60%2025.60%2C107.40%2030.10%2C110.10%2030.10%2C110.10%2067.30%2C132.20%2067.30%2C132.20%2069.50%2C133.50%2072.00%2C134.20%2074.50%2C134.20%2077.00%2C134.20%2079.50%2C133.50%2081.70%2C132.20%2081.70%2C132.20%20118.90%2C110.10%20118.90%2C110.10%20123.30%2C107.50%20126.10%2C102.60%20126.10%2C97.30%20126.10%2C97.30%20126.10%2C53.00%20126.10%2C53.00%20126.10%2C47.80%20123.40%2C42.90%20118.90%2C40.30%20Z)

And here's what I'm using in the config:

```json
{
    "bash": {
        "colour": "#000000",
        "icon": "<path d=\"M 89.40,81.10 C 89.70,80.90 89.90,81.10 90.00,81.50 90.00,81.50 90.00,84.70 90.00,84.70 91.30,84.20 92.50,84.00 93.60,84.30 93.80,84.40 93.90,84.70 93.80,85.00 93.80,85.00 93.10,87.80 93.10,87.80 93.00,88.00 92.90,88.20 92.80,88.40 92.70,88.50 92.70,88.50 92.60,88.50 92.50,88.50 92.40,88.60 92.30,88.50 91.80,88.40 90.70,88.10 88.90,89.10 87.00,90.10 86.30,91.70 86.40,92.90 86.40,94.40 87.20,94.80 89.70,94.80 93.10,94.90 94.60,96.40 94.70,99.80 94.70,103.20 92.90,106.90 90.10,109.10 90.10,109.10 90.20,112.30 90.20,112.30 90.20,112.70 90.00,113.10 89.70,113.30 89.70,113.30 87.80,114.40 87.80,114.40 87.50,114.60 87.30,114.40 87.20,114.00 87.20,114.00 87.20,110.90 87.20,110.90 85.60,111.60 84.00,111.70 82.90,111.30 82.70,111.20 82.60,110.90 82.70,110.60 82.70,110.60 83.40,107.70 83.40,107.70 83.50,107.50 83.60,107.20 83.70,107.10 83.80,107.00 83.80,107.00 83.90,107.00 84.00,106.90 84.10,106.90 84.20,107.00 85.30,107.40 86.80,107.20 88.10,106.50 89.90,105.60 91.00,103.80 91.00,102.00 91.00,100.40 90.10,99.70 88.00,99.70 85.30,99.70 82.80,99.20 82.70,95.20 82.70,91.90 84.40,88.50 87.10,86.40 87.10,86.40 87.10,83.20 87.10,83.20 87.10,82.80 87.30,82.40 87.60,82.20 87.60,82.20 89.40,81.10 89.40,81.10 Z M 109.20,101.50 C 109.20,101.50 99.90,107.00 99.90,107.00 99.70,107.10 99.50,107.30 99.50,107.60 99.50,107.60 99.50,110.00 99.50,110.00 99.50,110.30 99.70,110.40 99.90,110.30 99.90,110.30 109.30,104.60 109.30,104.60 109.50,104.50 109.60,104.20 109.60,103.90 109.60,103.90 109.60,101.80 109.60,101.80 109.60,101.50 109.40,101.40 109.20,101.50 Z M 118.90,40.30 C 118.90,40.30 81.70,18.20 81.70,18.20 79.50,16.90 77.00,16.20 74.50,16.20 72.00,16.20 69.50,16.90 67.30,18.20 67.30,18.20 30.10,40.30 30.10,40.30 25.70,42.90 22.90,47.80 22.90,53.10 22.90,53.10 22.90,97.30 22.90,97.30 22.90,102.60 25.60,107.40 30.10,110.10 30.10,110.10 67.30,132.20 67.30,132.20 69.50,133.50 72.00,134.20 74.50,134.20 77.00,134.20 79.50,133.50 81.70,132.20 81.70,132.20 118.90,110.10 118.90,110.10 123.30,107.50 126.10,102.60 126.10,97.30 126.10,97.30 126.10,53.00 126.10,53.00 126.10,47.80 123.40,42.90 118.90,40.30 Z M 68.60,129.80 C 68.60,129.80 31.40,107.70 31.40,107.70 27.80,105.50 25.50,101.50 25.50,97.20 25.50,97.20 25.50,53.00 25.50,53.00 25.50,48.70 27.80,44.60 31.40,42.50 31.40,42.50 68.60,20.40 68.60,20.40 70.40,19.30 72.40,18.80 74.50,18.80 76.60,18.80 78.60,19.40 80.40,20.40 80.40,20.40 117.60,42.50 117.60,42.50 120.70,44.30 122.70,47.50 123.30,51.00 122.10,48.40 119.30,47.60 116.00,49.50 116.00,49.50 80.90,71.30 80.90,71.30 76.50,73.90 73.30,76.70 73.30,82.00 73.30,82.00 73.30,125.40 73.30,125.40 73.30,128.60 74.60,130.60 76.50,131.20 75.90,131.30 75.20,131.40 74.50,131.40 72.50,131.50 70.40,130.90 68.60,129.80 Z M 118.90,40.30 C 118.90,40.30 81.70,18.20 81.70,18.20 77.30,15.60 71.80,15.60 67.30,18.20 67.30,18.20 30.10,40.30 30.10,40.30 25.70,42.90 22.90,47.80 22.90,53.10 22.90,53.10 22.90,97.30 22.90,97.30 22.90,102.60 25.60,107.40 30.10,110.10 30.10,110.10 67.30,132.20 67.30,132.20 69.50,133.50 72.00,134.20 74.50,134.20 77.00,134.20 79.50,133.50 81.70,132.20 81.70,132.20 118.90,110.10 118.90,110.10 123.30,107.50 126.10,102.60 126.10,97.30 126.10,97.30 126.10,53.00 126.10,53.00 126.10,47.80 123.40,42.90 118.90,40.30 Z\" />"
    }
}
```

What am I doing wrong? 

Thanks!

**Comment by mayurankv** [comment]
So sorry, I hadn't seen this! I'll have a play with it when I'm next doing some work on this and let you know!

**Reply by Cerebellum-ITM** [reply]
Hello, do you have an example of how to add an icon? or is there a way to do it by modifying the code? Thanks in advance

**Comment by leonardoperezl** [comment]
I'm also trying to do the same, I'll appreciate to get informed how to get such configuration done 😄

**Comment by mrkcmn** [comment]
Do you know of any database/library where i can find languages icons formatted like this?

**Reply by r4chl** [reply]
[https://devicon.dev](https://devicon.dev)

**Comment by mayurankv** [comment]
Planning to add a guide for this, will be added after the refactor.
